app/src/main/java/grinnell/appdev/edu/lyles/MenuFragment.java

```diff
@@ -60,7 +60,8 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         int recyclerViewId = (int) this.getArguments().get("recyclerViewId");
 
         View view = inflater.inflate(layoutXml, container, false);
-        mMenuItemArrayList = MenuItem.fromJSON(getMenuItemsAsJsonArray(url, arrayKey));
+        retrieveMenuData(url, arrayKey);
```

Comment:
Keep in mind, if this is asynchronous then the `mItemAdapter` created below will not have an actual value, so you should create it in the `onPostExecute` of your `AsyncRetrieval`